File: AMF5/src/test/java/javaPlatform/JsonLDSchemaParsedTest.java
package javaPlatform;

import amf.shapes.client.platform.JsonLDInstanceResult;
import amf.shapes.client.platform.JsonLDSchemaResult;
import amf.shapes.client.platform.config.JsonLDSchemaConfiguration;
import amf.shapes.client.platform.config.JsonLDSchemaConfigurationClient;
import amf.shapes.client.platform.model.domain.jsonldinstance.JsonLDObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.junit.Assert.assertTrue;

public class JsonLDSchemaParsedTest {
    @Test
    public void parseJsonSchema() throws ExecutionException, InterruptedException{
        final JsonLDSchemaConfigurationClient client = JsonLDSchemaConfiguration.JsonLDSchema().baseUnitClient();

        JsonLDSchemaResult jsonLDSchemaResult = client.parseJsonLDSchema("file://src/test/resources/jsonld-schemas/schema.json").get();
        JsonLDInstanceResult jsonLDInstanceResult = client.parseJsonLDInstance("file://src/test/resources/jsonld-schemas/instance.json", jsonLDSchemaResult.jsonDocument()).get();
        JsonLDObject jsonLDObject = (JsonLDObject) jsonLDInstanceResult.instance().encodes().get(0);
        assertTrue(jsonLDObject.graph().containsProperty("anypoint://vocabulary/security.yaml#sensitive"));
    }


}

File: AMF5/src/test/scala/scalaPlatform/JsonLDSchemaParsedTest.scala
package scalaPlatform

import amf.shapes.client.scala.config.{JsonLDSchemaConfiguration, JsonLDSchemaConfigurationClient}
import amf.shapes.client.scala.model.domain.jsonldinstance.JsonLDObject
import org.junit.runner.RunWith
import org.scalatest.funsuite.AsyncFunSuite
import org.scalatest.matchers.should.Matchers
import org.scalatestplus.junit.JUnitRunner

import scala.concurrent.ExecutionContext

@RunWith(classOf[JUnitRunner])
class JsonLDSchemaParsedTest extends AsyncFunSuite with Matchers{
  override implicit def executionContext: ExecutionContext = ExecutionContext.Implicits.global


  val client: JsonLDSchemaConfigurationClient =
    JsonLDSchemaConfiguration.JsonLDSchema().baseUnitClient()


  test("Parse JsonLDInstances with schema") {
    for {
      schema     <- client.parseJsonLDSchema(s"file://src/test/resources/jsonld-schemas/schema.json").map(_.jsonDocument)
      instance   <- client.parseJsonLDInstance(s"file://src/test/resources/jsonld-schemas/instance.json", schema)
    } yield {
      val graph = instance.instance.encodes.head.asInstanceOf[JsonLDObject].graph
      graph.containsProperty("anypoint://vocabulary/security.yaml#sensitive") shouldBe true
    }
  }
}
